src/lib.rs

```diff
@@ -71,9 +71,11 @@ mod srgb;
 pub mod text_metadata;
 mod traits;
 
+pub use crate::adam7::expand_pass as expand_interlaced_row;
 pub use crate::common::*;
 pub use crate::decoder::{
-    DecodeOptions, Decoded, Decoder, DecodingError, Limits, OutputInfo, Reader, StreamingDecoder,
+    DecodeOptions, Decoded, Decoder, DecodingError, InterlaceInfo, InterlacedRow, Limits,
```

Comment:
Possibly you should slightly bump up the doc comment on `InterlaceInfo`. The first sentence should probably state what it actually _is_.

Reply:
Also, maybe another commit with example code for how this new API should be used?

Reply:
I tried to improve the doc comments of `InterlaceInfo`.

> Also, maybe another commit with example code for how this new API should be used?

Ooops - I missed this.  Let me work on that.

Reply:
Ok, I've also added usage examples to the doc comments of `fn expand_interlaced_row`.  The example I used depended on being able to construct `Adam7Info`, so I also added a public `Adam7Info::new` function (which `asserts` that `pass` makes sense and panics otherwise).